test/sql/constraints/foreignkey/test_fk_concurrency_confilicts.test

```diff
@@ -0,0 +1,36 @@
+# name: test/sql/constraints/foreignkey/test_fk_concurrency_confilicts.test
+# description: FOREIGN KEY and concurency
+# group: [foreignkey]
+
+statement ok con1
+CREATE TABLE pk_integers(i INTEGER PRIMARY KEY)
+
+statement ok con1
+INSERT INTO pk_integers VALUES (1), (2), (3)
+
+statement ok con1
+BEGIN TRANSACTION
+
+statement ok con1
+CREATE TABLE fk_integers(j INTEGER, FOREIGN KEY (j) REFERENCES pk_integers(i))
+
+statement ok con1
+INSERT INTO fk_integers VALUES (1)
+
+statement error con2
+DROP TABLE fk_integers
```

Comment:
Could we add a test where this happens the other way around as well - i.e. the table is first dropped in one transaction and then another transaction adds the foreign key constraint.

Reply:
Okay, I will add it